components/app/leaderboard/FloatingVoteAction.vue

```diff
@@ -0,0 +1,79 @@
+<template>
+  <div
+    class="
+      fixed
+      w-full
+      flex
+      justify-end
+      lg:top-1/2
+      bottom-5
+      pointer-events-none
+    "
+  >
+    <app-link :to="voteUrl" :data-analytics="dataAnalytics">
+      <div
+        class="bg-white rounded-l shadow p-4 pointer-events-auto text-center"
+      >
+        <div
+          class="
+            bg-primary bg-opacity-10
+            relative
+            rounded
+            w-12
+            lg:w-16
+            h-12
+            lg:h-16
+            flex
+            items-center
+            justify-center
+            transition
+            duration-300
+            ease-in-out
+            lg:hover:scale-125
+          "
+        >
+          <app-icon class="text-primary" :size="36" :data="icons.vote" />
+        </div>
+
+        <div class="mt-1 text-primary">{{ $t('leaderboard.vote') }}</div>
+      </div>
+    </app-link>
+  </div>
+</template>
+
+<script>
```

Comment:
question(blocking): why are we using lang="ts" for some components and other are just js? If possible change this one to ts and add typehints

Reply:
The project configuration has no full TypeScript support yet. TypeScript compilation errors will occur if I am using vue properties. I am using TypeScript whenever possible. This is definitely a big TODO for OceanPearl@Nuxt3.